lib/descriptors/FieldDescriptor.js

```diff
@@ -131,6 +137,22 @@ FieldDescriptor.prototype.isRepeated = function () {
 }
 
 
+FieldDescriptor.prototype.setOneofIndex = function (oneofIndex) {
+  this._oneofIndex = oneofIndex
```

Comment:
it seems weird to me that they identify their oneofs by index instead of by name? would this break anything if the oneofs were moved around in the file?